Nodejs/Product/Nodejs/NodejsPackage.cs

```diff
@@ -180,31 +181,26 @@ private void MakeDebuggerContextAvailable()
             }
         }
 
-        internal IReplWindow2 OpenReplWindow(bool focus = true)
+        protected override int CreateToolWindow(ref Guid toolWindowType, int id)
         {
-            var compModel = this.ComponentModel;
-            var provider = compModel.GetService<IReplWindowProvider>();
-
-            var window = (IReplWindow2)provider.FindReplWindow(NodejsReplEvaluatorProvider.NodeReplId);
-            if (window == null)
+            if (toolWindowType == Guids.NodejsInteractiveWindow)
             {
-                window = (IReplWindow2)provider.CreateReplWindow(
-                    this.ReplContentType,
-                    Resources.InteractiveWindowTitle,
-                    Guids.TypeScriptLanguageInfo,
-                    NodejsReplEvaluatorProvider.NodeReplId
-                );
+                var model = (IComponentModel)GetService(typeof(SComponentModel));
+                var replProvider = model.GetService<InteractiveWindowProvider>();
```

Comment:
Consider extracting a method that returns the repl provider.

Reply:
yes